connect-inject/endpoints_controller.go

```diff
@@ -210,6 +223,111 @@ func (r *EndpointsController) SetupWithManager(mgr ctrl.Manager) error {
 		).Complete(r)
 }
 
+// getServiceCheck will return the health check for this pod and service if it exists.
+func getServiceCheck(client *api.Client, healthCheckID string) (*api.AgentCheck, error) {
+	filter := fmt.Sprintf("CheckID == `%s`", healthCheckID)
```

Comment:
I'm wondering if we may encounter a bug similar to the one where we weren't using `%q` when defining a filter in `connect-init` ? I'm guessing the `'s around "%s" make it work?

Reply:
yeah I think so, and I didn't notice issues testing it end to end, and that codepath would be used for updating the health check which I did test :)